### PR TITLE
Bump Base

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
 1. Addresses deprecations introduced in recent versions of `base`.
 2. Fixes visibility and documentation of public API related to composition of `Query` and `Topic`.
     Previously, the abstract base common to `QueryBuilder` and `TopicBuilder` wasn't a part of public API but was referenced in the public API documentation.

### CODE REVIEW

File: client/src/main/java/io/spine/client/QueryBuilder.java [hunk 1]
```diff
@@ -96,18 +96,16 @@ public QueryBuilder orderBy(String column, OrderBy.Direction direction) {
...
     public QueryBuilder limit(int count) {
         checkLimit(count);
         this.limit = count;
         return self();
     }
 
-    @SuppressWarnings("PMD.UnusedPrivateMethod")  /* See https://github.com/pmd/pmd/issues/770. */
-    private static void checkLimit(Number count) {
-        checkArgument(count.longValue() > 0, "A Query limit must be more than 0.");
+    static void checkLimit(long count) {
```
Comment (yuri-sergiichuk) [inline client/src/main/java/io/spine/client/QueryBuilder.java]: Let's add at least a one-liner doc here.

File: core/src/main/java/io/spine/core/Events.java [hunk 2]
```diff
@@ -93,10 +93,11 @@ public static EventMessage ensureMessage(Message eventOrMessage) {
...
      * @throws IllegalArgumentException
      *         if the ID string value is empty or blank
+     * @deprecated please use {@link io.spine.util.Preconditions2#checkNotDefaultArg(Message)}
      */
+    @Deprecated
     public static EventId checkValid(EventId id) {
-        checkNotNull(id);
-        checkNotEmptyOrBlank(id.getValue(), "event ID");
+        checkNotDefaultArg(id);
         return id;
```
Comment (yuri-sergiichuk) [inline core/src/main/java/io/spine/core/Events.java]: I believe we can just: `return checkNotDefaultArg(id);` here. But while it's deprecated already, not sure if it matters.

File: client/src/main/java/io/spine/client/TargetBuilder.java [hunk 3]
```diff
@@ -78,7 +78,7 @@
  * @param <B>
  *         a type of the builder implementations
  */
-abstract class AbstractTargetBuilder<T extends Message, B extends AbstractTargetBuilder> {
+public abstract class TargetBuilder<T extends Message, B extends TargetBuilder> {
```
Comment (yuri-sergiichuk) [inline client/src/main/java/io/spine/client/TargetBuilder.java]: So why is it public now?

File: server/src/main/java/io/spine/server/entity/storage/Columns.java [hunk 4]
```diff
@@ -114,17 +114,17 @@ static EntityColumn findColumn(Class<? extends Entity<?, ?>> entityClass, String
     }
 
     static void checkColumnName(String columnName) {
-        checkNotEmptyOrBlank(columnName, "entity column name");
+        checkNotEmptyOrBlank(columnName);
```
Comment (yuri-sergiichuk) [inline server/src/main/java/io/spine/server/entity/storage/Columns.java]: `Entity column name should not be empty or blank`?

File: server/src/main/java/io/spine/server/tenant/CurrentTenant.java [hunk 5]
```diff
@@ -76,8 +74,7 @@ static TenantId ensure() throws IllegalStateException {
      * @param tenantId a non-null and non-default instance of {@code TenantId}
      */
     static void set(TenantId tenantId) {
-        checkNotNull(tenantId);
-        checkArgument(isNotDefault(tenantId));
+        checkNotDefaultArg(tenantId);
```
Comment (yuri-sergiichuk) [inline server/src/main/java/io/spine/server/tenant/CurrentTenant.java]: `"Tenant ID should not be default"`?

File: server/src/test/java/io/spine/server/bus/BusesTest.java [hunk 6]
```diff
@@ -68,6 +71,12 @@ void passNullToleranceCheck() {
                                               .setCode(1)
                                               .build())
                 .setDefault(RejectionEnvelope.class, defaultRejection)
+                .setDefault(Message.class, StringValue.newBuilder()
```
Comment (yuri-sergiichuk) [inline server/src/test/java/io/spine/server/bus/BusesTest.java]: Will `TestValues.newUuidValue()` work here?

File: testutil-core/src/main/java/io/spine/testing/core/given/GivenTenantId.java [hunk 7]
```diff
@@ -52,9 +52,8 @@ public static TenantId generate() {
      * @param value must be non-null, not empty, and not-blank
      * @return new {@code TenantId}
      */
-    @SuppressWarnings("DuplicateStringLiteralInspection")
     public static TenantId of(String value) {
-        checkNotEmptyOrBlank(value, "value");
+        checkNotEmptyOrBlank(value);
```
Comment (yuri-sergiichuk) [inline testutil-core/src/main/java/io/spine/testing/core/given/GivenTenantId.java]: `"Tenant ID value should not be empty or blank."`?

File: client/src/main/java/io/spine/client/TargetBuilder.java (hunk 3 above)
Comment (alexander-yevsyukov) [reply client/src/main/java/io/spine/client/TargetBuilder.java]: Because we reference the class in Javadocs.

**Pull request merged**